File: dynamicDebug/src/main/java/com/cloud/dynamicdebug/MockUserModule.java
package com.cloud.dynamicdebug;

import com.cloud.disk.api.user.UserState;

import dagger.Binds;
import dagger.Module;
import dagger.hilt.InstallIn;
import dagger.hilt.android.components.ActivityComponent;

@Module
@InstallIn(ActivityComponent.class)
public abstract class MockUserModule {
    @Binds
    public abstract UserState bindUserState(
            MockUserStateImpl userStateImpl
    );
}

File: dynamicDebug/src/main/java/com/cloud/dynamicdebug/MockUserModule.kt
package com.cloud.dynamicdebug

import com.cloud.disk.api.user.UserState
import dagger.Binds
import dagger.Module
import dagger.hilt.InstallIn
import dagger.hilt.components.SingletonComponent

@Module
@InstallIn(SingletonComponent::class)
abstract class MockUserModule {
    @Binds
    abstract fun bindUserState(userStateImpl: MockUserStateImpl): UserState?
}